lib/ui/platform_dispatcher.dart

```diff
@@ -1314,6 +1371,134 @@ class WindowPadding {
+///
+/// ![Device with a hinge display feature](https://flutter.github.io/assets-for-api-docs/assets/hardware/display_feature_hinge.png)
+///
+/// ![Device with a fold display feature](https://flutter.github.io/assets-for-api-docs/assets/hardware/display_feature_fold.png)
+///
+/// ![Device with a cutout display feature](https://flutter.github.io/assets-for-api-docs/assets/hardware/display_feature_cutout.png)
+///
+/// The [state] contains information about the posture for foldable features
+/// ([DisplayFeatureType.hinge] and [DisplayFeatureType.fold]). The posture is
+/// the shape of the display, for example [DisplayFeatureState.postureFlat] or
+/// [DisplayFeatureState.postureHalfOpened]. For [DisplayFeatureType.cutout],
+/// the state is not used and has the [DisplayFeatureState.unknown] value.
+class DisplayFeature {
+  const DisplayFeature({
+    required this.bounds,
+    required this.type,
+    required this.state,
+  }) : assert(type != DisplayFeatureType.cutout || state == DisplayFeatureState.unknown);
+
+  /// The area of the flutter view occupied by this display feature, measured in logical pixels.
+  ///
+  /// On devices with two screens, the Flutter view spans from the top-left corner
+  /// of the left or top screen to the bottom-right corner of the right or bottom screen,
+  /// including the visual area occupied by any display feature. Bounds of display
+  /// features are reported in this coordinate system.
+  ///
+  /// For example, on a dual screen device in portrait mode:
+  ///
+  /// * [bounds.left] gives you the size of left screen, in logical pixels.
+  /// * [bounds.right] gives you the size of the left screen + the hinge width.
+  final Rect bounds;
+
+  /// Type of display feature, e.g. hinge, fold, cutout.
+  final DisplayFeatureType type;
+
+  /// Posture of display feature, which is populated only for folds and hinges.
+  ///
+  /// For cutouts, this is [DisplayFeatureState.unknown]
+  final DisplayFeatureState state;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is DisplayFeature && bounds == other.bounds &&
+        type == other.type && state == other.state;
+  }
+
+  @override
+  int get hashCode => hashValues(bounds, type, state);
+
+  @override
+  String toString() {
+    return 'DisplayFeature(rect: $bounds, type: $type, state: $state)';
+  }
+}
+
+/// Type of [DisplayFeature], describing the [DisplayFeature] behaviour and how
+/// much the it obstructs the display.
+///
+/// Some types of [DisplayFeature], like [DisplayFeatureType.fold], can be
+/// reported without actually impeding drawing on the screen. They are useful
+/// for knowing where the display is bent or has a crease. The
+/// [DisplayFeature.bounds] can be 0-width in such cases.
+///
+/// The shape of the display feature for types [DisplayFeatureType.fold] and
```

Comment:
"shape" may not be clear, you could make this clearer by mentioning one of the postures as an example